WMF Framework/ReadingListsSyncOperation.swift

```diff
@@ -754,21 +749,23 @@ internal class ReadingListsSyncOperation: ReadingListsOperation {
                     return
                 }
                 
-                guard let entry = NSEntityDescription.insertNewObject(forEntityName: "ReadingListEntry", into: moc) as? ReadingListEntry else {
-                    return
-                }
-                entry.update(with: remoteEntry)
-                if entry.createdDate == nil {
-                    entry.createdDate = NSDate()
-                }
-                if entry.updatedDate == nil {
-                    entry.updatedDate = entry.createdDate
-                }
-                entry.list = readingList
-                entry.articleKey = article.key
-                entry.displayTitle = article.displayTitle
-                if article.savedDate == nil {
-                    article.savedDate = entry.createdDate as Date?
+                if !readingList.articleKeys.contains(articleKey) {
```

Comment:
By the time we get to this function, the dupe check should have already occurred in `createOrUpdate(remoteReadingListEntries: [APIReadingListEntry], for readingListID: Int64? = nil, deleteMissingLocalEntries: Bool = false, inManagedObjectContext moc: NSManagedObjectContext)`. So I'd suggest keeping this function as it was before and seeing if the bug is one level up in `createOrUpdate`